src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/util/GraphQLRequestParamUtils.java

```diff
@@ -79,28 +80,24 @@ public static boolean isGraphQLContentType(final String contentType) {
      * @throws RuntimeException if JSON serialization fails for some reason due to any runtime environment issues
      */
     public static String toPostBodyString(final GraphQLRequestParams params) {
-        final ObjectMapper mapper = new ObjectMapper();
-        final ObjectNode postBodyJson = mapper.createObjectNode();
-        postBodyJson.set(OPERATION_NAME_FIELD,
-                JsonNodeFactory.instance.textNode(StringUtils.trimToNull(params.getOperationName())));
+        final StringBuilder sb = new StringBuilder(1024);
+
+        sb.append("{");
+
+        appendJsonStringField(sb, OPERATION_NAME_FIELD, StringUtils.trimToNull(params.getOperationName()));
+        sb.append(",");
 
         if (StringUtils.isNotBlank(params.getVariables())) {
-            try {
-                final ObjectNode variablesJson = mapper.readValue(params.getVariables(), ObjectNode.class);
-                postBodyJson.set(VARIABLES_FIELD, variablesJson);
-            } catch (JsonProcessingException e) {
-                log.error("Ignoring the GraphQL query variables content due to the syntax error: {}",
-                        e.getLocalizedMessage());
-            }
+            sb.append("\"").append(VARIABLES_FIELD).append("\":");
+            sb.append(StringUtils.trim(params.getVariables()));
+            sb.append(",");
         }
 
-        postBodyJson.set(QUERY_FIELD, JsonNodeFactory.instance.textNode(StringUtils.trim(params.getQuery())));
+        appendJsonStringField(sb, QUERY_FIELD, escapeNewLinesForJson(StringUtils.trim(params.getQuery())));
```

Comment:
Why only newlines are escaped? What if the value contains quotes, slashes, or something else?

Reply:
Pushed a new commit: a524a04fa69b40870e346ff805df51b7e5a4ba22 in order to use `JsonStringEncoder` to escape json string, which seems working fine for string including quotes, backslashes, new lines, etc.

Reply:
What do you think of using `JsonGenerator` to generate the json?

Reply:
Fixed using `JsonGenerator` for safety: 5c9fb4f92b20bb3feb5770023d7ed4ed021a29d6 Thanks!